x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/Range.java

```diff
@@ -108,10 +110,28 @@ public Object fold() {
 
     /**
      * Check whether the boundaries are invalid ( upper &lt; lower) or not.
-     * If they do, the value does not have to be evaluate.
+     * If they do, the value does not have to be evaluated.
      */
-    private boolean areBoundariesInvalid() {
-        Integer compare = BinaryComparison.compare(lower.fold(), upper.fold());
+    protected boolean areBoundariesInvalid() {
+        Object lowerValue = lower.foldable() ? lower.fold() : null;
```

Comment:
Is this ternary operator necessary here? The previous code just folded the values directly.

Reply:
Too defensive probably. I'm removing it.